Backend/Core/Services/UserService.cs

```diff
@@ -144,5 +144,29 @@ public async Task<string> GetUserIdByEmailAsync(string email)
 
             return user.Id;
         }
+
+        public async Task ChangePasswordAsync(ChangePasswordDTO changePasswordDTO, string userId)
+        {
+            if (changePasswordDTO.CurrentPassword == changePasswordDTO.NewPassword)
+            {
+                throw new HttpException(
+                        ErrorMessages.NewPasswordSamePrevious,
+                        HttpStatusCode.BadRequest);
+            }
+
+            var user = await _userRepository.GetByIdAsync(userId);
+
+            var result = await _userManager.ChangePasswordAsync(
+                user,
+                changePasswordDTO.CurrentPassword,
+                changePasswordDTO.NewPassword);
```

Comment:
Check whether the "CurrentPassword" was sent correctly, if not — return an error